e3sm_to_cmip/_logger.py

```diff
@@ -1,68 +1,75 @@
 
-def _setup_root_logger() -> str:  # pragma: no cover
-    """Sets up the root logger.
+def _logger(name=None, logfilename=DEFAULT_LOG, set_log_level=None, to_console=False, to_logfile=False, propagate=False):
+    """ Return a root or named logger with variable configuration.
 
-    The logger module will write to a log file and stream the console
-    simultaneously.
-
-    The log files are saved in a `/logs` directory relative to where
-    `e3sm_to_cmip` is executed.
-
-    Returns
-    -------
-    str
-        The name of the logfile.
+    Parameters
+    ----------
+    name : str
+        The name displayed for the logger in messages.
+        If name == None or name == "__main__", the root logger is returned
+    logfilename : str
+        If logfile handling is requested, any logfile may be specified, or else
+        the default (e2c_logs/dflt_log-{datetime.now(timezone.utc).strftime('%Y%m%d_%H%M%S_%f')}.log) is used.
+    set_log_level : str
+        One of { "DEBUG" (default), "INFO", "WARNING", "ERROR", "CRITICAL" }
+    to_console : boolean
+        If True, a logging.StreamHandler is supplied.  Default = False
+    to_logfile : boolean
+        If True, a logging.FileHandler is supplied. Default = False.
+    propagate : boolean
+        If True, messages logged are propagated to the root logger.  Default = False.
     """
-    os.makedirs("logs", exist_ok=True)
-    filename = f'logs/{UTC.localize(datetime.utcnow()).strftime("%Y%m%d_%H%M%S_%f")}'
-    log_format = "%(asctime)s_%(msecs)03d:%(levelname)s:%(funcName)s:%(message)s"
+  
+    if to_logfile:
+        dn = os.path.dirname(logfilename)
+        if len(dn) and not os.path.exists(dn):
+            os.makedirs(dn)
 
-    # Setup the logging module.
-    logging.basicConfig(
-        filename=filename,
-        format=log_format,
-        datefmt="%Y%m%d_%H%M%S",
-        level=logging.DEBUG,
-    )
-    logging.captureWarnings(True)
-    logging.Formatter.converter = time.gmtime
+    if name == None or name == "__main__":
+        logger = logger.root
+    else:
+        logger = logging.getLogger(name)
 
-    # Configure and add a console stream handler.
-    console_handler = logging.StreamHandler()
-    console_handler.setLevel(logging.INFO)
-    log_formatter = logging.Formatter(log_format)
-    console_handler.setFormatter(log_formatter)
-    logging.getLogger().addHandler(console_handler)
+    logger.propagate = propagate
 
-    return filename
+    if set_log_level == "None" or set_log_level == "DEBUG":
+        log_level = DEFAULT_LOG_LEVEL
+    elif set_log_level == "INFO":
+        log_level = logging.INFO
+    elif set_log_level == "WARNING":
+        log_level = logging.WARNING
+    elif set_log_level == "ERROR":
+        log_level = logging.ERROR
+    elif set_log_level == "CRITICAL":
+        log_level = logging.CRITICAL
+    else: log_level = DEFAULT_LOG_LEVEL
```

Comment:
We agreed to remove this logic and make `set_log_level` the `logging` module levels instead of string?

Reply:
Also suggest renaming `set_log_level` to `log_level`.

Reply:
As mentioned earlier - each of the handler modules would need to "import logging", just to be able to pass the defined values "logging.WARNING", "logging.INFO", etc.  I can do that.  - I don't suppose any significant overhead is incurred in having each handler "import" another module ("logging", on top of our "_logger.py").  It would certainly make "_logger" cleaner.